benchmarks/src/main/java/io/druid/benchmark/query/GroupByBenchmark.java

```diff
@@ -0,0 +1,343 @@
+  public void setup2() throws IOException
+  {
+
+  }
+
+  @TearDown(Level.Iteration)
+  public void teardown() throws IOException
+  {
+  }
+
+  private IncrementalIndex makeIncIndex()
+  {
+    return new OnheapIncrementalIndex(
+        0,
+        QueryGranularity.NONE,
+        QueryBenchmarkUtil.INGESTION_AGGS,
+        true,
+        false,
+        true,
+        MAX_ROWS
+    );
+  }
+
+  public static final QuerySegmentSpec intervalSpec = new MultipleIntervalSegmentSpec(
+      Arrays.asList(QUERY_INTERVAL)
+  );
+
+  private static final GroupByQuery query = GroupByQuery
+      .builder()
+      .setDataSource("blah")
+      .setQuerySegmentSpec(intervalSpec)
+      .setDimensions(Lists.<DimensionSpec>newArrayList(
+          new DefaultDimensionSpec("dimA", null),
+          new DefaultDimensionSpec("dimB", null)
+          //new DefaultDimensionSpec("dimC", null),
+          //new DefaultDimensionSpec("dimD", null)
+          //new DefaultDimensionSpec("dimE", null), //multival dims greatly increase the running time, disable for now
+          //new DefaultDimensionSpec("dimF", null)
+          ))
+      .setAggregatorSpecs(
+          QUERY_AGGS
+      )
+      .setGranularity(QueryGranularity.DAY)
+      .build();
+
+  private static <T> List<T> runQuery(QueryRunnerFactory factory, QueryRunner runner, Query<T> query)
+  {
+
+    QueryToolChest toolChest = factory.getToolchest();
+    QueryRunner<T> theRunner = new FinalizeResultsQueryRunner<>(
+        toolChest.mergeResults(toolChest.preMergeQueryDecoration(runner)),
+        toolChest
+    );
+
+    Sequence<T> queryResult = theRunner.run(query, Maps.<String, Object>newHashMap());
+    return Sequences.toList(queryResult, Lists.<T>newArrayList());
+  }
+
+
+  @Benchmark
+  @BenchmarkMode(Mode.AverageTime)
+  @OutputTimeUnit(TimeUnit.MICROSECONDS)
+  public void queryIncrementalIndex(Blackhole blackhole) throws Exception
+  {
+    QueryRunner<Row> runner = QueryBenchmarkUtil.makeQueryRunner(
+        factory,
+        "incIndex",
+        new IncrementalIndexSegment(incIndex, "incIndex")
+    );
+
+    List<Row> results = GroupByBenchmark.runQuery(factory, runner, query);
+
+    for (Row result : results) {
+      blackhole.consume(result);
+    }
+  }
+
+  @Benchmark
+  @BenchmarkMode(Mode.AverageTime)
+  @OutputTimeUnit(TimeUnit.MICROSECONDS)
+  public void queryQueryableIndex(Blackhole blackhole) throws Exception
+  {
+    QueryRunner<Row> runner = QueryBenchmarkUtil.makeQueryRunner(
+        factory,
+        "qIndex",
+        new QueryableIndexSegment("qIndex", qIndex)
+    );
+
+    List<Row> results = GroupByBenchmark.runQuery(factory, runner, query);
+
+    for (Row result : results) {
+      blackhole.consume(result);
+    }
+  }
+
+
+  // Currently redundant, only merges one segment, and runQuery() already calls toolchest.mergeResults()
+  // Kept here for reference, if in the future a benchmark that merges results from multiple runners is added.
```

Comment:
took out the dead code, the query benchmarks have a multi-segment/thread benchmark now